questions.js

```diff
@@ -1,74 +1,79 @@
-//    jsonData = JSON5.parse(data);
-//    getDataFromJson(jsonData, questions);
-//    console.log(questionsJS.questions);
-//});
+    jsonData = JSON5.parse(data);
+
+    getDataFromJson(jsonData);
+
+    jf.writeFile('generatedCountries.json', countries);
+
+});
 
-for(var i in originCountries) {
-    correctNameEnding = changeCountryNameEnding(originCountries[i]);
-    questions.push(correctNameEnding);
-}
 
-function changeCountryNameEnding(country) {
-    var countryLength = country.length,
-        endings = country[countryLength-1],
-        nameWithoutEnding = country.slice(0, countryLength-1),
-        correctName;
-
-    switch(endings) {
-        case 'а':
-            correctName = nameWithoutEnding+'и';
-            break;
-        case 'я':
-            correctName = nameWithoutEnding+'ї';
-            break;
-        case 'й':
-            correctName = nameWithoutEnding+'я';
-            break;
-        case 'н':
-            correctName = nameWithoutEnding+'на';
-            break;
-        case 'р':
-            correctName = nameWithoutEnding+'ра';
-            break;
-        case 'с':
-            correctName = nameWithoutEnding+'са';
-            break;
-        case 'ь':
-            correctName = nameWithoutEnding+'і';
-            break;
-        case 'м':
-            correctName = nameWithoutEnding+'ма';
-            break;
-        case 'и':
-            correctName = nameWithoutEnding;
-            break;
-        default:
-            correctName = country;
+function countryNameToGenitiveForm (country) {
+
+    var countryEnding = country[country.length-1],
+        countryWithoutEnding = country.slice(0, country.length - 1),
+        newCountryForm;
+
+    for(var i in consonantals){
+
+        if(countryEnding === consonantals[i] && countryEnding !== 'й' && countryEnding !== 'ш' && countryEnding !== 'з'){
+            newCountryForm = country + 'у';
+        }
+        else if(countryEnding === 'й'){
+            newCountryForm = countryWithoutEnding + "ю";
+        }
+        else if(countryEnding === 'ш'){
+            newCountryForm = country;
+        }
+        else if(countryEnding === 'з'){
+            newCountryForm = country + 'а';
+        }
     }
 
-    return correctName;
+    if(countryEnding === 'я'){
+        newCountryForm = countryWithoutEnding + 'ї';
+    }
+    else if(countryEnding === 'а'){
+        newCountryForm = countryWithoutEnding + 'и';
+    }
+    else if(countryEnding === 'і' || countryEnding === 'о' || countryEnding === 'у' || countryEnding === 'е'){
+        newCountryForm = country;
+    }
+
+    return newCountryForm;
 }
 
-//function getDataFromJson(json, question) {
-//	for(var i in json){
-//		if(json[i].capital !== null) {
-//			question.push({"name": json[i].name, "capital": json[i].capital});
-//		}
-//		else {
-//			question.push({"name": json[i].name, "capital": json[i].name});
-//		}
-//	}
-//}
+function getDataFromJson(json) {
```

Comment:
Can you read https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map and think if you can rewrite code using `map` and `filter` function instead of `for` loop and `if` statements. This is just exercise for you to try. I'm going to merge this PR.